frame_utils.go

```diff
@@ -0,0 +1,75 @@
+package tchannel
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+	"sync"
+)
+
+type RecordingFramePool struct {
+	sync.Mutex
```

Comment:
https://github.com/uber-go/guide/blob/master/style.md#avoid-embedding-types-in-public-structs

especially don't want to embed a mutex